kafka/service.go

```diff
@@ -43,11 +43,19 @@ func (s *Service) CreateAndTestClient(ctx context.Context, l *zap.Logger, opts [
 	}
 
 	// Test connection
-	connectCtx, cancel := context.WithTimeout(ctx, 15*time.Second)
-	defer cancel()
-	err = s.testConnection(client, connectCtx)
-	if err != nil {
-		return nil, fmt.Errorf("failed to test connectivity to Kafka cluster %w", err)
+	for {
+		err = s.testConnection(client, ctx)
+		if err == nil {
+			break
+		}
+		returnErr := fmt.Errorf("failed to test connectivity to Kafka cluster %w", err)
+
+		if !s.cfg.RetryInitConnection {
+			return nil, returnErr
+		}
+
+		logger.Error(fmt.Sprintf("%s, retrying in 5 seconds", returnErr.Error()))
```

Comment:
```suggestion
		logger.Error("retrying in 5 seconds",  zap.Error(err)))
```

Maybe also worth to degrade to warning as it may be in the category of "expected" if users are using retry init?